kong/core/resolver.lua

```diff
@@ -1,64 +1,39 @@
 local singletons = require "kong.singletons"
 local url = require "socket.url"
 local cache = require "kong.tools.database_cache"
-local stringy = require "stringy"
 local constants = require "kong.constants"
 local responses = require "kong.tools.responses"
 
 local table_insert = table.insert
-local string_match = string.match
-local string_find = string.find
-local string_format = string.format
-local string_sub = string.sub
-local string_gsub = string.gsub
-local string_len = string.len
 local ipairs = ipairs
-local unpack = unpack
 local type = type
 
 local _M = {}
 
 -- Take a request_host and make it a pattern for wildcard matching.
 -- Only do so if the request_host actually has a wildcard.
 local function create_wildcard_pattern(request_host)
-  if string_find(request_host, "*", 1, true) then
-    local pattern = string_gsub(request_host, "%.", "%%.")
-    pattern = string_gsub(pattern, "*", ".+")
-    pattern = string_format("^%s$", pattern)
-    return pattern
+  if request_host:find("*", 1, true) then
+    return "^"..request_host:gsub("%.", "%%."):gsub("*", ".+").."$"
   end
 end
 
 -- Handles pattern-specific characters if any.
 local function create_strip_request_path_pattern(request_path)
-  return string_gsub(request_path, "[%(%)%.%%%+%-%*%?%[%]%^%$]", function(c) return "%"..c end)
+  return request_path:gsub("[%(%)%.%%%+%-%*%?%[%]%^%$]", function(c) return "%"..c end)
```

Comment:
This involved a lookup on such a hot code path that `gsub(subject, ...)` did not. Same for all of the string functions you removed in this module.